server/test/services/export/PdfExporterTest.java

```diff
@@ -30,10 +35,23 @@ public void validatePDFExport() throws IOException, DocumentException {
     int pages = pdfReader.getNumberOfPages();
     for (int i = 1; i < pages; i++) {
       textFromPDF.append(PdfTextExtractor.getTextFromPage(pdfReader, i));
+      // Assertions to check if the URL is embedded for the FileUpload
+      PdfDictionary pdfDictionary = pdfReader.getPageN(i);
+      PdfArray annots = pdfDictionary.getAsArray(PdfName.ANNOTS);
+      PdfObject current = annots.getPdfObject(0);
+      PdfDictionary currentPdfDictionary = (PdfDictionary) pdfReader.getPdfObject(current);
+      assertThat(currentPdfDictionary.get(PdfName.SUBTYPE)).isEqualTo(PdfName.LINK);
+      PdfDictionary AnnotationAction = (PdfDictionary) currentPdfDictionary.getAsDict(PdfName.A);
+      assertThat(AnnotationAction.get(PdfName.S)).isEqualTo(PdfName.URI);
+      PdfString link = AnnotationAction.getAsString(PdfName.URI);
+      assertThat(link.toString())
+          .isEqualTo(
+              "http://localhost:9000/admin/programs/"
+                  + applicationOne.getProgram().id
+                  + "/files/my-file-key");
     }
 
     pdfReader.close();
-
     assertThat(textFromPDF).isNotNull();
```

Comment:
Is there any browser test coverage of this? I understand that asserting PDF content in a browser test would likely not be worth our time. That said, we could have a test that clicks the download button and asserts that a non-empty file is downloaded.

As it stands, we have a bit of a gap in our coverage for a scenario where the PDFExporter code properly generates the data, but for some reason our controller doesn't send it back to the client.

Reply:
Added a browser tests